scripts/util/resolve-types.js

```diff
@@ -125,8 +124,8 @@ module.exports = (shapesWithoutNames) => {
 
   resolve.structureCategory = (sh) => {
     if (sh.exception) { return 'exception'; }
-    if (sh.name.endsWith('Request')) { return 'request'; }
-    if (sh.name.endsWith('Response')) { return 'response'; }
+    if (outputShapes.indexOf(sh.name) !== -1) { return 'response'; }
+    if (inputShapes.indexOf(sh.name) !== -1) { return 'request'; }
```

Comment:
Request and response types are now determined based on whether or not they are input and/or output to an operation. Note that in at least one case, a type is both input and output. Since we don't declare input (request) types, but need to declare output types, output takes precedence.